tests/s3_data_plane_tests.c

```diff
@@ -1294,6 +1294,72 @@ static int s_test_s3_put_object_with_part_remainder(struct aws_allocator *alloca
     return 0;
 }
 
+AWS_TEST_CASE(test_s3_get_object_presigned_url, s_test_s3_get_object_presigned_url)
+static int s_test_s3_get_object_presigned_url(struct aws_allocator *allocator,void *ctx) {
+    (void)ctx;
+    struct aws_s3_tester tester;
+    AWS_ZERO_STRUCT(tester);
+    ASSERT_SUCCESS(aws_s3_tester_init(allocator, &tester));
+    struct aws_string *presigned_url_string = aws_string_new_from_cursor(allocator, &g_test_presigned_url);
+
+    if (presigned_url_string->len == 0) {
+        AWS_LOGF_INFO(
+            AWS_LS_S3_GENERAL, "The presigned url environment variable was not set. Skip this test.");
+        aws_string_destroy(presigned_url_string);
+        aws_s3_tester_clean_up(&tester);
+        return 0;
+    }
+    struct aws_s3_client_config client_config = {
+        .part_size = 64 * 1024,
+    };
+    ASSERT_SUCCESS(aws_s3_tester_bind_client(
+        &tester, &client_config, AWS_S3_TESTER_BIND_CLIENT_REGION | AWS_S3_TESTER_BIND_CLIENT_SIGNING));
+    struct aws_s3_client *client = aws_s3_client_new(allocator, &client_config);
+
+    struct aws_http_message *message = aws_http_message_new_request(allocator);
+    ASSERT_TRUE(message != NULL);
+
+    char host_header_value[1024];
+    snprintf(
+        host_header_value,
+        sizeof(host_header_value),
+        "%.*s.s3.us-west-2.amazonaws.com",
+        (int)g_test_bucket_name.len,
+        g_test_bucket_name.ptr);
+
+    struct aws_http_header host_header =
+        {
+            .name = g_host_header_name,
+            .value = aws_byte_cursor_from_c_str(host_header_value),
+        };
+
+    ASSERT_SUCCESS(aws_http_message_add_header(message, host_header));
+    ASSERT_SUCCESS (aws_http_message_set_request_method(message, aws_http_method_get));
+    ASSERT_SUCCESS (aws_http_message_set_request_path(message, g_test_presigned_url));
+
+    struct aws_s3_meta_request_options options;
+    AWS_ZERO_STRUCT(options);
+    options.type = AWS_S3_META_REQUEST_TYPE_GET_OBJECT;
+    options.message = message;
+    options.presigned_url = aws_string_c_str(presigned_url_string);
```

Comment:
So, you can test like:
```suggestion
    struct aws_signing_config_aws signing_config;
    AWS_ZERO_STRUCT(signing_config);
    aws_s3_init_default_signing_config(&signing_config, g_test_s3_region, NULL);
    struct aws_credentials *credentials = aws_credentials_new_anonymous(allocator); 
    signing_config.credentials = credentials;
    options.signing_config = &signing_config;
    
```

Reply:
I changed to this with a header <aws/auth/credentials.h> and a release `aws_credentials_release(credentials);`, but it threw the following error:
```
6: Fatal error condition occurred in /Users/haozhao/Downloads/simurgh/aws-c-s3/source/s3_util.c:278: credentials_provider
46: Exiting Application
46: [INFO] [2023-06-07T16:57:13Z] [000000016fa53000] [event-loop] - id=0x158104110: default timeout 100s, and max events to process per tick 100
46: [TRACE] [2023-06-07T16:57:13Z] [000000016fa53000] [event-loop] - id=0x158104110: waiting for a maximum of 100s 0ns
46: ################################################################################
46: Stack trace:
46: ################################################################################
46: 1   aws-c-s3-tests                      0x0000000100b01c8c aws_backtrace_print + 188
46: 2   aws-c-s3-tests                      0x0000000100adcdac aws_fatal_assert + 104
46: 3   aws-c-s3-tests                      0x0000000100a2af18 aws_s3_init_default_signing_config + 112
46: 4   aws-c-s3-tests                      0x00000001009cf3e0 s_test_s3_get_object_presigned_url + 1740
46: 5   aws-c-s3-tests                      0x00000001009c45c8 s_aws_run_test_case + 336
46: 6   aws-c-s3-tests                      0x00000001009c4b9c test_s3_get_object_presigned_url + 32
46: 7   aws-c-s3-tests                      0x00000001009b14a0 main + 1364
46: 8   dyld                                0x0000000100fd508c start + 520
1/1 Test #46: test_s3_get_object_presigned_url ...Subprocess aborted***Exception:   0.26 sec

```
it pointed to s3_util.c line 278 `void aws_s3_init_default_signing_config` so I assume the credential provider is not allowed to be `Null`?

Also, I think for our use case, we also need the logic that skipping signing config whenever presigned_url was identified. If not, we have to pass the "empty" signing config from sdk cpp or crt cpp side and I am not sure that is workable due to the untouchable Client class.

Reply:
well... Then.
```suggestion
    struct aws_signing_config_aws signing_config;
    AWS_ZERO_STRUCT(signing_config);
    struct aws_credentials_provider *provider = aws_credentials_provider_new_anonymous(allocator, NULL);
    aws_s3_init_default_signing_config(&signing_config, g_test_s3_region, provider);
    options.signing_config = &signing_config;
```

All I wanted to say is you can skip signing without changing aws-c-s3. :)

Reply:
```
struct aws_signing_config_aws signing_config;
AWS_ZERO_STRUCT(signing_config);
struct aws_credentials_provider_shutdown_options shutdown_options;
AWS_ZERO_STRUCT(shutdown_options);

struct aws_credentials_provider *provider = aws_credentials_provider_new_anonymous(allocator, &shutdown_options);
aws_s3_init_default_signing_config(&signing_config, g_test_s3_region, provider);
options.signing_config = &signing_config;
```
works but I didn't know whether this way of skipping signing config is something the team allows.